stix_shifter_modules/guardium/stix_transmission/guard_utils.py

```diff
@@ -108,15 +108,16 @@ def validate_response(self, p_response, prefix, abort=False):
             return False
         return True    
 
-    def handle_report(self, report_name, params, index_from, fetch_size):
+    def handle_report(self, params, index_from, fetch_size):
         # -------------------------------------------------------------------------------
         # REPORT
         # -------------------------------------------------------------------------------
         results = ""
         #context().logger.debug('-------------------  ' + report_name + ' ----------------------')
-        params_set = {"reportName":"{0}".format(report_name), "indexFrom": "{0}".format(index_from), "fetchSize": "{0}".format(fetch_size), 
-        "reportParameter":params, "inputTZ":"UTC"}
-        json_dump = json.dumps(params_set)
+        params["fetchSize"] = int(fetch_size)
+        params["index_from"]=int(index_from)
+        params["inputTZ"]="UTC"
+        json_dump = json.dumps(params)
         rest_data = str(json.loads(json_dump))
```

Comment:
```
       json_dump = json.dumps(params)
         rest_data = str(json.loads(json_dump))
```
I don't think you need to dump the dictionary object and then load and again cast to string. Seems like some redundant operations. You may be able to use the `params` in the request method.

Reply:
reportName and reportParameter are put into the query in the query construction step, I removed redundant code that was reassembling the query again before execution

Reply:
Removed redundant json conversion